oeps/oep-0044-bp-changelogs.rst

```diff
@@ -0,0 +1,212 @@
+* It is both acceptable and appropriate to clean up or fix changelog entries at any time.
+
+Types of Changes
+~~~~~~~~~~~~~~~~
+
+These are useful words to start your changelog entries.
+
+* **Add** for new features.
+* **Change** for changes in existing functionality.
+* **Deprecate** for soon-to-be removed features.
+* **Remove** for now removed features, usually prefixed with **BREAKING CHANGE:**.
+* **Fix** for any bug fixes.
+
+Include **BREAKING CHANGE** for all backward incompatible changes.
+The Security Working Group may want to manage timing when using: **Security** in case of vulnerabilities.
+
+Example Changelog
+~~~~~~~~~~~~~~~~~
+
+Here is an example CHANGELOG.rst for a repository that contains unversioned code.
+
+::
+
+  Changelog
+  ===========
+
+  All notable changes to this project will be documented in this file.
+  The format is based on recommendations in `OEP-44: Changelogs`_.
+
+  .. _`OEP-44: Changelogs`: https://open-edx-proposals.readthedocs.io/en/latest/oep-0044-bp-changelogs.html
+
+  2020-03-21
+  ----------
+
+  Optional summary text for this date's entry.
+
+  * **BREAKING CHANGE**: Remove deprecated function XXX. See DEPR-XXX for details.
+
+    * Optionally use sub-bullet for additional text or links about this change.
+
+  * **BREAKING CHANGE**: Remove deprecated feature YYY.
+
+
+  2020-03-14
+  ----------
+
+  * Added feature x. See announcement here: https://xxx
+  * Deprecated function XXX. See DEPR-XXX for details.
+
+For a repository with versioned code, use the version in the headers like::
+
+  [1.0.0] - 2020-03-21
+  --------------------
+
+* You may include an optional section at the top to collect changes that aren't yet released, although we typically release every change
+
+::
+
+  [Unreleased]
+  ------------
+
+Commit Message vs Changelog Entry
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Since changelog messages are for developers and consumers of your code, good changelog messages will often not match commit messages. Here are some example updated messages.
+
+.. list-table::
+   :header-rows: 1
+   :widths: 50 50
+
+   * - Commit Message
+     - Changelog Entry
+   * - deps: update dependency css-loader to v3.5.1
+     - No functional change
+
+       Note: This example changelog entry assumes there were no other changes for this version.
+   * - Fix SyntaxError in UserLogout class
+     - Fix 500 error during logout
+   * - Rename dry_run parameter
+     - **BREAKING CHANGE** Remove the dry_run parameter in the public foobarize API method. This parameter is deprecated in favour of the no_apply parameter. See docs for details.
+   * - Add set_foobarizer method to api.Foo
+     - Add a set_foobarizer method to Foo's public API. This is particularly useful for developers trying to foobarize their users. See docs for details.
+
+Github Releases
+~~~~~~~~~~~~~~~
+
+* Github Releases should not replace the CHANGELOG.rst.
+* If used, Github Releases should stay in sync with the CHANGELOG.rst.
```

Comment:
FYI: Github adds "Latest Release" next to the latest Github Release.  So, if newer tags just go to CHANGELOG.rst, it is very misleading.  Possible solutions:
1. Transfer all release information to CHANGELOG.rst and delete all Github Releases.  Note: hopefully no one adds a new Github Release again.  Is automation required to ensure that releases don't exist?
2. Publish a release with title "See CHANGELOG.rst" and body "See CHANGELOG.rst" for every new release.  Annoying and ugly, but works.